tensorflow/lite/experimental/micro/examples/mnist_demo/main.cc

```diff
@@ -0,0 +1,140 @@
+      return float32Str;
+    case kTfLiteInt32:
+      return int32Str;
+    case kTfLiteUInt8:
+      return uInt8Str;
+    case kTfLiteInt64:
+      return int64Str;
+    case kTfLiteString:
+      return stringStr;
+    case kTfLiteBool:
+      return boolStr;
+    case kTfLiteInt16:
+      return int16Str;
+    case kTfLiteComplex64:
+      return complex64Str;
+    case kTfLiteInt8:
+      return int8Str;
+    default:
+      return noTypeStr;
+  }
+}
+
+// Helper fn to log the shape and datatype of a tensor
+void printTensorDetails(TfLiteTensor* tensor,
+                        tflite::ErrorReporter* error_reporter) {
+  error_reporter->Report("Type [%s] Shape :", typeString(tensor->type));
+  for (int d = 0; d < tensor->dims->size; ++d) {
+    error_reporter->Report("%d [ %d]", d, tensor->dims->data[d]);
+  }
+  error_reporter->Report("");
+}
+
+int main(int argc, char* argv[]) {
+  // Set up logging.
+  tflite::MicroErrorReporter micro_error_reporter;
+  tflite::ErrorReporter* error_reporter = &micro_error_reporter;
+
+  // Map the model into a usable data structure. This doesn't involve any
+  // copying or parsing, it's a very lightweight operation.
+  error_reporter->Report("Parsing MNIST classifier model FlatBuffer.");
+  const tflite::Model* model = ::tflite::GetModel(mnist_dense_model_tflite);
+  if (model->version() != TFLITE_SCHEMA_VERSION) {
+    error_reporter->Report(
+        "Model provided is schema version %d not equal "
+        "to supported version %d.\n",
+        model->version(), TFLITE_SCHEMA_VERSION);
+    return 1;
+  }
+  error_reporter->Report("Model parsed.\n");
+
+  // This pulls in all the operation implementations we need.
+  tflite::ops::micro::AllOpsResolver resolver;
+
+  // Create an area of memory to use for input, output, and intermediate arrays.
+  // The size of this will depend on the model you're using, currently
+  // determined by experimentation.
+  const int tensor_arena_size = 200 * 1024;
+  uint8_t tensor_arena[tensor_arena_size];
+
+  // Build an interpreter to run the model with.
+  error_reporter->Report("Build interpreter.");
+  tflite::MicroInterpreter interpreter(model, resolver, tensor_arena,
+                                       tensor_arena_size, error_reporter);
+  interpreter.AllocateTensors();
+  error_reporter->Report("Done.\n");
+
+  // Get information about the models input and output tensors.
+  TfLiteTensor* model_input = interpreter.input(0);
+  error_reporter->Report("Details of input tensor:");
+  printTensorDetails(model_input, error_reporter);
+  TfLiteTensor* model_output = interpreter.output(0);
+  error_reporter->Report("Details of output tensor:");
+  printTensorDetails(model_output, error_reporter);
+
+  // perform inference on each test sample and evalute accuracy of model
+  int accurateCount = 0;
+  for (int s = 0; s < mnistSampleCount; ++s) {
+    // Set value of input tensor
+    for (int d = 0; d < 784; ++d) model_input->data.f[d] = mnistInput[s][d];
```

Comment:
Prefer declaring a const variable along with a comment explaining what this constant is for.  Also, shouldn't this value be 28 * 28 = 784 instead of 748?

Additionally, our linter requires:
for (...) {
    ...
}

Reply:
I've added a const int initialized to 28 * 28 now. It's much clearer and is the correct number!